tests/utils/CMakeLists.txt

```diff
@@ -1,36 +1,26 @@
 # This file is part of covfie, a part of the ACTS project
 #
-# Copyright (c) 2022 CERN
+# Copyright (c) 2022-2023 CERN
 #
 # This Source Code Form is subject to the terms of the Mozilla Public License,
 # v. 2.0. If a copy of the MPL was not distributed with this file, You can
 # obtain one at http://mozilla.org/MPL/2.0/.
 
-find_package(
-    Boost
-    1.71.0
-    REQUIRED
-    COMPONENTS
-    filesystem
-)
-
 add_library(
     testing_utils
 
     tmp_file.cpp
 )
 
-# Ensure that the utils.
-target_link_libraries(
-    testing_utils
-
-    PUBLIC
-    Boost::filesystem
-)
-
 target_include_directories(
     testing_utils
     PUBLIC
 
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
 )
+
+target_compile_definitions(
+    testing_utils
+    PRIVATE
+    _CRT_SECURE_NO_WARNINGS
+)
```

Comment:
More my curiosity than anything else but what does this do?

Reply:
It is connected to `std::tmpnam`. That function is really discouraged by both MSVC and GCC. 😦

```
  tmp_file.cpp
C:\Users\krasz\ATLAS\covfie\covfie\tests\utils\tmp_file.cpp(20,25): warning C4996: 'tmpnam': This function or variable may be unsafe. Consider using tmpnam_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [C:\Users\krasz\ATLAS\covfie\build\tests\utils\testing_utils.vcxproj]
```

With GCC there is no compile-time warning coming from the usage of this function, but at linkt-ime we sure get one. 😦

```
/usr/bin/ld: ../utils/libtesting_utils.a(tmp_file.cpp.o): in function `get_tmp_file[abi:cxx11]()':
tmp_file.cpp:(.text+0x2e): warning: the use of `tmpnam' is dangerous, better use `mkstemp'
```

Unfortunately to make both MSVC and GCC happy, we would need to write platform specific implementations for them. Which I didn't think was warranted for this function. As it's really not super important for this to be super safe.

Reply:
Okay, fair enough. Thanks!
=